### ISSUE TITLE
SpaceNet is not checking length consistency btw X and y

### ISSUE THREAD

**Issue by ahoyosid** [nilearn/nilearn#1134]
SpaceNet is not checking_consistent_length(X, y) when using the whole brain.

On the example "plot_haxby_space_net.py"

### Using the whole brain: It won't raise an error ###
```python
# Remove one sample from y_train
y_train = y_train[:-1]

decoder = SpaceNetClassifier(memory="nilearn_cache", penalty='graph-net',
                             screening_percentile=100)
decoder.fit(X_train, y_train)

y_pred = decoder.predict(X_test)
accuracy = (y_pred == y_test).mean() * 100.
print("Graph-net classification accuracy : %g%%" % accuracy)
```
Graph-net classification accuracy : 97.7778%

### Using screening: This will raise an error because it uses SelectPercentile ###
```python 
decoder = SpaceNetClassifier(memory="nilearn_cache", penalty='graph-net',
                             screening_percentile=20)
decoder.fit(X_train, y_train)

y_pred = decoder.predict(X_test)
accuracy = (y_pred == y_test).mean() * 100.
print("Graph-net classification accuracy : %g%%" % accuracy)
```
ValueError: Found arrays with inconsistent numbers of samples: [125 126]

I added a backport to scikit-learn validation module, which I'm currently using on #698. This can be added easily to solve this issue.

Should I split #698  on two different PRs?, one for the backports and another for the Decoder

**Comment by bthirion** [comment]
I think you should.  Thx for pointing this.

**Comment by dohmatob** [comment]
On Fri, Jun 17, 2016 at 11:11 AM, Andrés Hoyos Idrobo <
notifications@github.com> wrote:

> SpaceNet is not checking_consistent_length(X, y) when using the whole
> brain.
>
> On the example "plot_haxby_space_net.py"
> Using the whole brain: It won't raise an error
>
> # Remove one sample from y_train
> y_train = y_train[:-1]
>
> decoder = SpaceNetClassifier(memory="nilearn_cache", penalty='graph-net',
>                              screening_percentile=100)
> decoder.fit(X_train, y_train)
>
> y_pred = decoder.predict(X_test)
> accuracy = (y_pred == y_test).mean() * 100.print("Graph-net classification accuracy : %g%%" % accuracy)
>
> Graph-net classification accuracy : 97.7778%
> Using screening: This will raise an error because it uses SelectPercentile
>
> decoder = SpaceNetClassifier(memory="nilearn_cache", penalty='graph-net',
>                              screening_percentile=20)
> decoder.fit(X_train, y_train)
>
> y_pred = decoder.predict(X_test)
> accuracy = (y_pred == y_test).mean() * 100.print("Graph-net classification accuracy : %g%%" % accuracy)
>
> ValueError: Found arrays with inconsistent numbers of samples: [125 126]
>
Could you please post a minimalistic code example which reproduces the bug
and the complete traceback thereof ? Thanks in advancE.

> I added a backport to scikit-learn validation module, which I'm currently
> using on #698 <https://github.com/nilearn/nilearn/pull/698>. This can be
> added easily to solve this issue.
>
> Should I split #698 <https://github.com/nilearn/nilearn/pull/698> on two
> different PRs, one for the backports and another for the Decoder ?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nilearn/nilearn/issues/1134>, or mute the thread
> <https://github.com/notifications/unsubscribe/AAms1MK-IoKxLooNuMuztmA_qt-GIBMNks5qMmS8gaJpZM4I4K68>
> .
>



-- 
Elvis Dohmatob,
*sent from my  potato...*
http://dohmatob.blogspot.fr/, https://team.inria.fr/parietal/elvis/,
https://github.com/dohmatob

**Comment by AlexandreAbraham** [comment]
Fixed by #1148.
